File: evd_ros_backend/evd_ros_core/src/evd_script/program/control_flow/breakpoint.py
from ..primitive import Primitive


class Breakpoint(Primitive):

    '''
    Data structure methods
    '''

    @classmethod
    def type_string(cls, trailing_delim=True):
        return 'breakpoint' + '.' if trailing_delim else ''

    @classmethod
    def full_type_string(cls):
        return Primitive.full_type_string() + cls.type_string()

    def __init__(self, type='', name='', uuid=None, parent=None, append_type=True):
        super(Breakpoint,self).__init__(
            type=Breakpoint.type_string() + type if append_type else type,
            name=name,
            uuid=uuid,
            parent=parent,
            append_type=append_type)

    '''
    Execution methods
    '''

    def symbolic_execution(self, hooks):
        pass

    def realtime_execution(self, hooks):
        pass

File: evd_authoring_app/src/model/evdScript/program/control_flow/breakpoint.js
import { Primitive } from '../primitive';

export class Breakpoint extends Primitive {

    /*
     * Data structure methods
     */

     static typeString() {
         return 'breakpoint.';
     }

     static fullTypeString() {
         return Primitive.fullTypeString() + Breakpoint.typeString();
     }

     constructor(type='', name='', uuid=null, parent=null, appendType=true) {
         super(
            (appendType) ? 'breakpoint.'+type : type,
            name,
            uuid,
            parent,
            appendType
         );
     }

    static fromDict(dct) {
        return new Breakpoint(
            dct.type,
            dct.name,
            dct.uuid,
            null,
            false
        );
    }

    static BlocklyToolbox() {
        return { type: 'breakpoint' };
    }

    static BlocklyBlock() {
        return { key: 'breakpoint', data: {
            init: function() {
                this.appendDummyInput()
                    .appendField("Breakpoint");
                this.setPreviousStatement(true, null);
                this.setNextStatement(true, null);
                this.setColour(160);
                this.setTooltip("breakpoint");
                this.setHelpUrl("breakpoint");
            }
        }};
    }
}